hail/src/main/scala/is/hail/expr/types/physical/PInterval.scala

```diff
@@ -9,123 +9,55 @@ import is.hail.utils._
-        }
-      }
-    }
-
-  def endPrimaryUnsafeOrdering(): UnsafeOrdering =
-    new UnsafeOrdering {
-      private val pOrd = pointType.unsafeOrdering()
-      def compare(r1: Region, o1: Long, r2: Region, o2: Long): Int = {
-        val edef1 = endDefined(r1, o1)
-        if (edef1 == endDefined(r2, o2)) {
-          val cmp = pOrd.compare(r1, loadEnd(r1, o1), r2, loadEnd(r2, o2))
-          if (cmp == 0) {
-            val includesE1 = includesEnd(r1, o1)
-            if (includesE1 == includesEnd(r2, o2)) {
-              val sdef1 = startDefined(r1, o1)
-              if (sdef1 == startDefined(r2, o2)) {
-                val cmp = pOrd.compare(r1, loadStart(r1, o1), r2, loadStart(r2, o2))
-                if (cmp == 0) {
-                  val includesS1 = includesStart(r1, o1)
-                  if (includesS1 == includesStart(r2, o2)) {
-                    0
-                  } else if (includesS1) 1 else -1
-                } else cmp
-              } else if (sdef1) -1 else 1
-            } else if (includesE1) -1 else 1
-          } else cmp
-        } else {
-          if (edef1) -1 else 1
-        }
-      }
-    }
-
-  val representation: PStruct = PStruct(
-      required,
-      "start" -> pointType,
-      "end" -> pointType,
-      "includesStart" -> PBooleanRequired,
-      "includesEnd" -> PBooleanRequired)
-
-  def startOffset(off: Code[Long]): Code[Long] = representation.fieldOffset(off, 0)
-
-  def endOffset(off: Code[Long]): Code[Long] = representation.fieldOffset(off, 1)
-
-  def loadStart(region: Region, off: Long): Long = representation.loadField(region, off, 0)
-
-  def loadStart(region: Code[Region], off: Code[Long]): Code[Long] = representation.loadField(region, off, 0)
-
-  def loadStart(rv: RegionValue): Long = loadStart(rv.region, rv.offset)
-
-  def loadEnd(region: Region, off: Long): Long = representation.loadField(region, off, 1)
-
-  def loadEnd(region: Code[Region], off: Code[Long]): Code[Long] = representation.loadField(region, off, 1)
-
-  def loadEnd(rv: RegionValue): Long = loadEnd(rv.region, rv.offset)
-
-  def startDefined(region: Region, off: Long): Boolean = representation.isFieldDefined(region, off, 0)
-
-  def endDefined(region: Region, off: Long): Boolean = representation.isFieldDefined(region, off, 1)
-
-  def includesStart(region: Region, off: Long): Boolean = Region.loadBoolean(representation.loadField(region, off, 2))
-
-  def includesEnd(region: Region, off: Long): Boolean = Region.loadBoolean(representation.loadField(region, off, 3))
-
-  def startDefined(off: Code[Long]): Code[Boolean] = representation.isFieldDefined(off, 0)
-
-  def endDefined(off: Code[Long]): Code[Boolean] = representation.isFieldDefined(off, 1)
-
-  def includeStart(off: Code[Long]): Code[Boolean] =
-    Region.loadBoolean(representation.loadField(off, 2))
-
-  def includeEnd(off: Code[Long]): Code[Boolean] =
-    Region.loadBoolean(representation.loadField(off, 3))
+  def copy(required: Boolean): PInterval
```

Comment:
put the unsafe ordering back here, I think

Reply:
endPrimaryUnsafeOrdering as well?

Reply:
yes. This only uses `pointType`.